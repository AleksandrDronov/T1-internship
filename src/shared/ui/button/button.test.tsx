/* eslint-disable no-alert */
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";
import { Button } from ".";

describe("Button renders correctly", () => {
  test("default", () => {
    const tree = renderer.create(<Button>Submit</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("disabled", () => {
    const tree = renderer
      .create(<Button disabled>Disabled Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("primary", () => {
    const tree = renderer
      .create(<Button variant="primary">Primary Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("secondary", () => {
    const tree = renderer
      .create(<Button variant="secondary">Secondary Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("tertiary", () => {
    const tree = renderer
      .create(<Button variant="tertiary">Tertiary Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("link", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Button isLink to="/test">
            Link
          </Button>
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("onClick", () => {
    window.alert = jest.fn();
    const onClick = () => alert("Button clicked");
    render(<Button onClick={onClick}>Button</Button>);
    const button = screen.getByText("Button");
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith("Button clicked");
  });
});
