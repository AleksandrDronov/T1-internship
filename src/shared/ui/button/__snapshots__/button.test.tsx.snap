// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly default 1`] = `
<button
  className="button primary"
  disabled={false}
  type="button"
>
  Submit
</button>
`;

exports[`Button renders correctly disabled 1`] = `
<button
  className="button primary disabled"
  disabled={true}
  type="button"
>
  Disabled Button
</button>
`;

exports[`Button renders correctly link 1`] = `
<a
  className="button primary link"
  href="/test"
  onClick={[Function]}
>
  Link
</a>
`;

exports[`Button renders correctly primary 1`] = `
<button
  className="button primary"
  disabled={false}
  type="button"
>
  Primary Button
</button>
`;

exports[`Button renders correctly secondary 1`] = `
<button
  className="button secondary"
  disabled={false}
  type="button"
>
  Secondary Button
</button>
`;

exports[`Button renders correctly tertiary 1`] = `
<button
  className="button tertiary"
  disabled={false}
  type="button"
>
  Tertiary Button
</button>
`;
